code/datums/components/server.dm

```diff
@@ -0,0 +1,64 @@
+// Server component
+// will generate a heat based on the power usage of the machine
+// use only with /obj/machinery
+
+
+
+/datum/component/server
+
+	var/efficiency = 1 // 0 to 1 range
+
+	var/temperature = T20C // current temperature
+	var/warning_temp = T0C + 50 // 50C
+	var/overheated_temp = T0C + 100 // 100C - temperature at which the server stops working
+	var/heat_capacity = 2500 // Used for auxmos heat transfer
+
+	var/heat_generation = 0 // heat generated by the machine
+
+	// if it sparks when overheated
+	var/sparks = TRUE
+	COOLDOWN_DECLARE(spark_cooldown)
+	var/sparks_cooldown_time = 10 SECONDS
+
+/datum/component/server/Initialize()
+	if(!ismachinery(parent)) // currently only compatible with machinery
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_MACHINERY_POWER_USED, PROC_REF(ParentPowerUsed))
+	START_PROCESSING(SSservers, src)
```

Comment:
there is no `STOP_PROCESSING()` for this on del
but also no `unregister signal` too